x/merkledb/trieview.go

```diff
@@ -161,31 +161,27 @@ func newTrieView(
 	}
 
 	for _, op := range changes.BatchOps {
-		newVal := maybe.Nothing[[]byte]()
 		key := op.Key
+		if changes.ConsumeBytes {
```

Comment:
Should this be `!changes.ConsumeBytes`? 

```go
	// ConsumeBytes when set to true will skip copying of bytes and assume
	// ownership of the provided bytes.
```
but we're not skipping copying here